Bottom: 867fbf689dbc7fe2866d50a5cfb6d1faaf30b502
Top:    ddfb7e42932e32ff6f79f0e7bd4d6ab292d5bc3d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-14 01:27:29 +0100

hid/gtk: Remove unused gc->erase parameter from GDK and GL renderers


---

diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index 8130f15..48d9fc2 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -51,7 +51,6 @@ typedef struct hid_gc_struct
   gint width;
   gint cap, join;
   gchar xor_mask;
-  gchar erase;
   gint mask_seq;
 }
 hid_gc_struct;
@@ -381,12 +380,10 @@ ghid_set_color (hidGC gc, const char *name)
   if (strcmp (name, "erase") == 0)
     {
       gdk_gc_set_foreground (gc->gc, &gport->bg_color);
-      gc->erase = 1;
     }
   else if (strcmp (name, "drill") == 0)
     {
       gdk_gc_set_foreground (gc->gc, &gport->offlimits_color);
-      gc->erase = 0;
     }
   else
     {
@@ -425,8 +422,6 @@ ghid_set_color (hidGC gc, const char *name)
 	{
 	  gdk_gc_set_foreground (gc->gc, &cc->color);
 	}
-
-      gc->erase = 0;
     }
 }
 
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 2d80bb2..95e0f11 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -56,7 +56,6 @@ typedef struct hid_gc_struct
   gint width;
   gint cap, join;
   gchar xor;
-  gchar erase;
 }
 hid_gc_struct;
 
@@ -374,14 +373,12 @@ ghid_set_color (hidGC gc, const char *name)
     gport->colormap = gtk_widget_get_colormap (gport->top_window);
   if (strcmp (name, "erase") == 0)
     {
-      gc->erase = 1;
       r = gport->bg_color.red   / 65535.;
       g = gport->bg_color.green / 65535.;
       b = gport->bg_color.blue  / 65535.;
     }
   else if (strcmp (name, "drill") == 0)
     {
-      gc->erase = 0;
       alpha_mult = 0.85;
       r = gport->offlimits_color.red   / 65535.;
       g = gport->offlimits_color.green / 65535.;
@@ -428,8 +425,6 @@ ghid_set_color (hidGC gc, const char *name)
       r = cc->red;
       g = cc->green;
       b = cc->blue;
-
-      gc->erase = 0;
     }
   if (1) {
     double maxi, mult;
